sensor_msgs/msg/CompressedImage.msg

```diff
@@ -38,4 +41,10 @@ string format                # Specifies the format of the data
                              #   need for successful decoding of the image. Refer to
                              #   documentation of the other transports for details.
 
+string compression_type      # Compression type used (jpeg, png, theora, etc)
+
+uint64 sequence_number       # sequence number
+uint64 flags                 # flags (for example: KEYFRAME)
```

Comment:
Again, I don't think I agree that this should be here.  A compressed image doesn't represent anything other than data, and I don't think we should be adding "external" metadata like this.

Reply:
In general, this argument is about whether there should be *any* metadata fields in the message, or if the compression plugin should just shove its metadata into the binary blob in some bespoke fashion. To the extent that a common set of metadata exists, I think it is preferable to put that metadata in the message so that message introspection tooling can see it, rather than putting this all in binary blob form and practically inaccessible.

Reply:
> In general, this argument is about whether there should be _any_ metadata fields in the message

I disagree.  I absolutely think we should have metadata in this message that is intrinsic to the `CompressedImage`.  Things like `width`, `height`, `format`, `compression_type`, and `is_bigendian` all belong in here.

But metadata about the larger context in which this data is delivered doesn't seem to belong, in my opinion.  The exception to this is of course `std_msgs/Header`, which is already here and is heavily used with the rest of our tooling.

I definitely understand the need for things like a sequence_number, a keyframe flag, and the like.  But those seem like things that should be in a custom message, composed of a `CompressedImage` plus the metadata, like:

```
# CompressedKeyFrame
sensor_msgs/CompressedImage compressed_image
int sequence_number
```

Reply:
> But those seem like things that should be in a custom message, composed of a CompressedImage plus the metadata...

Frankly, I really think that if the outcome isn't an improved-for-everyone message, the right answer is for `image_transport` to just define its own internal message type that it can change on its own. I don't think that's better for the community, but if we have to introduce a new type for `image_transport` uses, having it inherit all the inertia that accompanies being a part of core message types isn't an improvement either.

Reply:
I think that Chris is right that the higher level information such as keyframes and sequence numbers deserve to be in a higher level concept than just the compressed image. Things like Key Frames and sequences are only valid if they're associated with a specific stream. Where a standalone compressed image may or may not be used in those applications. 

In particular if someone muxes two compressed Image topics together, the sequence numbers become meaningless unless you're associating them with a name stream or the like. 

There's the approach above of using a hierarchy of messages/wrapper messages. The other alternative would be to send a parallel stream of messages in the same way that we do with `camera_info`. There's a parallel stream of camera info which can be associated with the image by timestamp + frame_id and with that this extra metadata can be applied in the image_transport process, but can be ignored by those who just want to access the raw images. And those images should standalone, in log files etc. Thus there would be CompressedImages and then there would be a CompressedImageStream protocol/standard for how to send along a series of compressed images with keyframes and potentially other metadata in a parallel `CompressedImageStreamInfo` or the like.